device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsCbsReceiverLinkHandler.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.device.transport.amqps;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.amqp.messaging.Properties;
+import org.apache.qpid.proton.amqp.messaging.Released;
+import org.apache.qpid.proton.amqp.transport.DeliveryState;
+import org.apache.qpid.proton.engine.Delivery;
+import org.apache.qpid.proton.engine.Event;
+import org.apache.qpid.proton.engine.Receiver;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Every SAS token based authentication over AMQP requires a CBS session with a sender and receiver link. This
+ * class defines the receiver link which receives authentication status codes corresponding to each authentication attempt.
+ */
+@Slf4j
+public final class AmqpsCbsReceiverLinkHandler extends AmqpsReceiverLinkHandler
+{
+    private final static String APPLICATION_PROPERTY_STATUS_CODE = "status-code";
+    private final static String APPLICATION_PROPERTY_STATUS_DESCRIPTION = "status-description";
+
+    private static final String RECEIVER_LINK_ENDPOINT_PATH = "$cbs";
+    private static final String RECEIVER_LINK_TAG_PREFIX = "cbs-receiver";
+
+    private Map<UUID, AuthenticationMessageCallback> correlationMap = new ConcurrentHashMap<>();
```

Comment:
what is correlationmap?

Reply:
I can add some comments to explain this. When the CBS receiver link gets a message from iothub, it needs to correlate that message's correlation id back to a callback so that a device session knows that it was authenticated